app/assets/javascripts/lock.js

```diff
@@ -0,0 +1,39 @@
+$(function () {
+  var datetimeFormat = 'MM/DD/YYYY h:mm A';
+  $('.datetimepicker').datetimepicker();
+
+  function formatLockExpirationTime() {
+    $('.new-lock-form').submit(function (e) {
+      var lockDeleteAtInput = $(e.target).find('#lock_delete_at');
+      var lockDeleteAtVal = lockDeleteAtInput.val();
+
+      if (lockDeleteAtVal !== '') {
+        var lockExpireMoment = moment(lockDeleteAtVal, datetimeFormat);
+        lockDeleteAtInput.val(lockExpireMoment.utc().format());
+      }
+    });
+  }
+
+  function handleLockTimeButtons() {
+    $('.lock-times').click(function (e) {
+      var buttonData = e.target.dataset;
+      var unit = buttonData.unit;
+      var quantity = parseInt(buttonData.num);
+      var lockDeleteAtInput = $(this).siblings('.datetimepicker-container').find('#lock_delete_at');
```

Comment:
having 2 different ways of getting the lock_delete_at seems a bit strange ...

maybe iterate over the $('.new-lock-form') ?

```js
$('.new-lock-form').each -> e
  delete_at = foo
  $(e).submit() stuff with delete_at
  $(e, .lock-times).click() stuff with delete_at
```